Check table names in upper-camel-lower-title cases

coresh opened the pull request:
Mysql / MariaDB / Persona allows to use the table name in different cases.

The table names might look like this:

UPPERCASETABLENAME
CamelCaseTableName
lowercasetablename
Titlecasetablename

However, the original implementation of the tablesExist() function searches for existing tables ONLY in lower case:
by using the function:
'strtolower()'

$names = array_map('strtolower', $names);

return count($names) === count(array_intersect($names, $tableNames));

The current pull request allows to search for table names in different registers. Optional argument "$lowerCase = true" - allows to maintain compatibility with previous versions

Thank you

<!-- Fill in the relevant information below to help triage your pull request. -->

|      Q       |   A
|------------- | -----------
| Type         | bug/feature/improvement
| Fixed issues | <!-- use #NUM format to reference an issue -->

#### Summary

<!-- Provide a summary of your change. -->

coresh committed:
Check table names in upper-camel-lower-title cases

Mysql / MariaDB / Persona allows to use the table name in different cases.

The table names might look like this:

UPPERCASETABLENAME
CamelCaseTableName
lowercasetablename
Titlecasetablename

However, the original implementation of the tablesExist() function
searches for existing tables ONLY in lower case:
by using the function:
'strtolower()'

$names = array_map('strtolower', $names);

return count($names) === count(array_intersect($names, $tableNames));

The current pull request allows to search for table names in different registers.
Optional argument "$lowerCase = true" - allows to maintain compatibility with previous versions

Thank you

morozov commented on src/Schema/AbstractSchemaManager.php:
```diff
@@ -309,7 +309,7 @@ protected function doListTableIndexes($table): array
      * @throws Exception
      */
-    public function tablesExist($names)
+    public function tablesExist($names, $lowerCase = true)
```
This doesn't look like the right solution. When it comes to object names, the DBAL should behave the same as the underlying database without the user having to provide any additional input.

coresh replied:
Re: "DBAL should behave the same as the underlying database":
DBAL will be the same as the underlying database without the user having to provide any additional input.
That's why the argument "$lowerCase" is added as an optional parameter: $lowerCase = true and can be ignored.

Means:

Can be used in two scenarios:

Scenario "A": $this->tablesExist([UserNames, Credentials], false);
Scenario "B": $this->tablesExist([UserNames, Credentials]);

"A" and "B" - both scenarios properly records. DBAL / Object names: safe. 

Can the function be implemented without optional argument "$lowerCase = true":
Correct: can be.

But, can you imagine how many projects can be
who used exactly this feature - naming tables in lower case.

The main goal of the optional argument: "$lowerCase = true" is:
give backwards compatibility for projects, who used exactly this feature - naming tables in lower case.

Thank you

coresh committed:
Merge branch 'doctrine:3.6.x' into 3.6.x

coresh committed:
Allow table names in upper-camel-lower-title cases

Updated pull request

Mysql / MariaDB / Persona allows to use the table name in different cases.

The table names might look like this:

UPPERCASETABLENAME
CamelCaseTableName
lowercasetablename
Titlecasetablename

However, the original implementation of the tablesExist() function searches for existing tables ONLY in lower case:
by using the function:
'strtolower()'

$names = array_map('strtolower', $names);

return count($names) === count(array_intersect($names, $tableNames));

The current pull request allows to search for table names in different registers/cases - upper-camel-lower-title-mixed.

Reverse compatibility - observed.
The function will give the correct result for any table names registers/cases - upper-camel-lower-title-mixed.

Thank you

coresh commented:
The request is closed.

The reason:

There is a more elegant solution

coresh commented:
@morozov 

Thanks for focusing on thе issue

Updated pull request:

[https://github.com/doctrine/dbal/pull/5922#issue-1583746183](https://github.com/doctrine/dbal/pull/5922#issue-1583746183)
